smart_open/tests/test_smart_open.py

```diff
@@ -483,6 +483,22 @@ def test_s3_mode_mock(self, mock_write, mock_boto):
         mock_boto.connect_s3().get_bucket.assert_called_with("mybucket")
         self.assertTrue(mock_write.called)
 
+    @mock.patch('smart_open.smart_open_lib.boto')
+    @mock.patch('smart_open.smart_open_lib.S3OpenWrite')
+    def test_s3_unsecured_mode_mock(self, mock_write, mock_boto):
+        """Are s3u:// open modes passed correctly?"""
+        # Configure the mock boto.config.get to return default host
+        smart_open.smart_open_lib.boto.config.get.return_value = 'example.com'
+
+        # correct write mode, correct s3 URI
+        smart_open.smart_open("s3u://user:secret@example.com@mybucket/mykey", "w")
+        mock_boto.connect_s3.assert_called_with(aws_access_key_id='user', aws_secret_access_key='secret',
```

Comment:
Prefer hanging indent, not vertical alignment ([pep8 examples](https://www.python.org/dev/peps/pep-0008/#indentation)).